### ISSUE TITLE
wmiexec cmd exec method is not being used by default

### ISSUE THREAD

**Issue by Bruk0ut** [byt3bl33d3r/CrackMapExec#224]
## Steps to reproduce

1. On current stable Kali release (3.1.5 Smidge) or Bleeding Edge (4.0.1dev - Bug Pr0n), invoke the -x argument to execute a command, either without specifying the exec method (it should use wmiexec by default), or by explicitly specifying the 'wmiexec' exec method as per the attached screenshots.

## Command string used
1) current stable (Kali): crackmapexec 192.168.1.240 -d w7-test1 -u administrator -H <hash> -x calc.exe
and
2) bleeding edge: ./cme smb 192.168.1.240 -d w7-test1 -u administrator -H <hash> -x calc.exe --exec-method wmiexec

## CME verbose output (using the --verbose flag)
1) current stable (Kali):
DEBUG                                          {'domain': 'w7-test1', 'wdigest': None, 'verbose': True, 'sam': False, 'cred_id': [], 'module_options': [], 'fail_limit': None, 'share': 'C$', 'lusers': False, 'module': None, 'smb_port': 445, 'show_options': False, 'rid_brute': None, 'uac': False, 'ufail_limit': None, 'pass_pol': False, 'regex': None, 'list_modules': False, 'no_output': False, 'pattern': None, 'lsa': False, 'force_ps32': False, 'shares': False, 'content': False, 'server_host': '0.0.0.0', 'wmi': None, 'exclude_dirs': '', 'server_port': None, 'wmi_namespace': '//./root/cimv2', 'gfail_limit': None, 'mssql_query': None, 'username': ['administrator'], 'hash': ['<redacted>'], 'users': False, 'sessions': False, 'exec_method': None, 'spider': None, 'ps_execute': None, 'threads': 100, 'mssql_port': 1433, 'password': [], 'mssql': False, 'mssql_auth': 'windows', 'ntds_pwdLastSet': False, 'execute': 'calc.exe', 'target': ['192.168.1.240'], 'ntds_history': False, 'disks': False, 'ntds': None, 'server': 'https', 'depth': 10, 'local_auth': False, 'timeout': 20}
[*] KTHXBYE!

2) bleeding edge:
DEBUG Passed args:
{'clear_obfscripts': False,
 'content': False,
 'cred_id': [],
 'darrell': False,
 'depth': None,
 'disks': False,
 'domain': 'w7-test1',
 'exclude_dirs': '',
 'exec_method': 'wmiexec',
 'execute': 'calc.exe',
 'fail_limit': None,
 'force_ps32': False,
 'gen_relay_list': None,
 'gfail_limit': None,
 'groups': None,
 'hash': ['<redacted>'],
 'jitter': None,
 'list_modules': False,
 'local_auth': False,
 'local_groups': None,
 'loggedon_users': False,
 'lsa': False,
 'module': None,
 'module_options': [],
 'no_output': False,
 'ntds': None,
'obfs': False,
 'only_files': False,
 'pass_pol': False,
 'password': [],
 'pattern': None,
 'port': 445,
 'protocol': 'smb',
 'ps_execute': None,
 'regex': None,
 'rid_brute': None,
 'sam': False,
 'server': 'https',
 'server_host': '0.0.0.0',
 'server_port': None,
 'sessions': False,
 'share': 'C$',
 'shares': False,
 'show_module_options': False,
 'spider': None,
 'spider_folder': '.',
 'target': ['192.168.1.240'],
 'threads': 100,
 'timeout': None,
 'ufail_limit': None,
 'username': ['administrator'],
 'users': None,
'verbose': True,
 'wmi': None,
 'wmi_namespace': 'root\\cimv2'}

## CME Version (cme --version)
Kali release (3.1.5 Smidge)
Bleeding Edge (4.0.1dev - Bug Pr0n)

## OS
Kali 2017.3

## Target OS
Windows 7 x64

## Detailed issue explanation
As per https://github.com/byt3bl33d3r/CrackMapExec/wiki/Command-Execution, cme should default to using WMI first for executing commands as per the failover methods.
In a lab environment, TCP/445 (SMB) was intentionally blocked between the Kali VM running cme and Windows 7 64bit target to ensure WMI would be used.

While monitoring network activity when executing commands however, it was observed that WMI is never attempted by cme and that it actually tries SMB first, even when explicitly specifying to use the wmiexec method via the command line args.

It was expected to see connectivity over TCP/135 (for WMI) however cme never tries, and due to the intentional firewall blocking TCP/445 during these tests, the cme commands failed.

Please see the attached screenshots.

Cheers

Bruk0ut
![cme1](https://user-images.githubusercontent.com/5972704/34465010-38cbb90c-ee4f-11e7-8711-dbf442aaed5b.png)
![cme2](https://user-images.githubusercontent.com/5972704/34465011-38f37e06-ee4f-11e7-9061-0626cdd9640b.png)
![cme3](https://user-images.githubusercontent.com/5972704/34465012-3906bdf4-ee4f-11e7-8bbf-7fdcf69f3be7.png)

**Comment by byt3bl33d3r** [comment]
So I think what you're seeing here is the initial connection that get's made to the host over SMB so that it can pull the host information (OS, architecture, domain, etc...). The reasoning behind this is because we need the domain name the machine is joined to to authenticate the DCOM connection to execute a command via WMI. That's why CME requires SMB to be enabled to execute WMI commands (or really anything that goes over DCOM). Now, in theory we could execute a WMI command without first connecting to the target via SMB and just specify all the information on the command line, but this would require a pretty extensive re-write since this is just how CME works right now. Furthermore, I have yet to see an environment that blocks SMB but leaves the RPC port open, and I'm pretty sure this isn't a common scenario so I don't think it would be worth the effort to accomadate this edge-case (but i'm definitely open to being corrected).

TL;DR this is an edge case which would require a extensive overhaul to correct, I'm closing this and marking it as a won't-fix for now. Feel free to comment below if I'm misunderstanding something.

**Comment by ama21n** [comment]
Hi I’d also be very keen to see this. Would provide a very cool way to run commands over 135 when RDP/SMB are blocked. I’ve come across this edge case a number of times and the standard WMI tools usually use 445 first also
